src/Stryker.Core/Stryker.Core/Mutants/CsharpNodeOrchestrators/CommentParser.cs

```diff
@@ -65,32 +66,55 @@ private static MutationContext ParseStrykerComment(MutationContext context, Matc
         return context.FilterMutators(disable, filteredMutators, match.Groups[OnceGroup].Value.ToLower() == "once", comment);
     }
 
-    public static MutationContext ParseNodeComments(SyntaxNode node, MutationContext context)
+    public static MutationContext ParseNodeLeadingComments(SyntaxNode node, MutationContext context)
     {
-        foreach (var commentTrivia in node.GetLeadingTrivia()
-                     .Where(t => t.IsKind(SyntaxKind.SingleLineCommentTrivia) ||
-                                 t.IsKind(SyntaxKind.MultiLineCommentTrivia)).Select(t => t.ToString()))
+        // parse comment that is on a dedicated line before the syntx node
+        if (node.GetLeadingTrivia()
+            .Where(t => t.IsKind(SyntaxKind.SingleLineCommentTrivia) ||
+                        t.IsKind(SyntaxKind.MultiLineCommentTrivia)).Select(t => t.ToString()).Any(commentTrivia => ProcessComment(node, ref context, commentTrivia)))
         {
-            // perform a quick pattern check to see if it is a 'Stryker comment'
-            if (!Pattern.Match(commentTrivia).Success)
-            {
-                continue;
-            }
-
-            var match = Parser.Match(commentTrivia);
-            if (match.Success)
-            {
-                // this is a Stryker comments, now we parse it
-                context = ParseStrykerComment(context, match, node);
-                break;
-            }
+            return context;
```

Comment:
I notice that this skips the rest of the logic if any leading trivia are present. Is that intentional? Would we want to allow for cases where, for whatever reason, someone is using both trailing trivia on the previous token and leading trivia on the current token?

This would certainly be a strange way for someone to write their code, but I would still expect Stryker's comment parsing logic to handle it as one would expect, unless supporting it would break other stuff:
```csharp
public bool DoSomeLogic(int a, int b)
{   // Stryker disable arithmetic
    // Stryker disable equality
    return a + b < 10;
}
```

Reply:
Indeed, having multiple Stryker comments is currently undefined behavior. 
There is currently no scenario where it is needed. One can provide a list of mutators if one needs to disable or enable more than one. That is, you can do this:
`// Stryker disable arithmetic,equality`

Support for multi line comments offers new way to haves several Stryker comments for the same syntax node. But it does not provide any valid reason to do so.

Current behavior is that Stryker stops processing after the first valid Stryker comment.
Possible behaviors are:
1. Aggregate all Stryker comments, with the following logic
   1. Sequentially, that is each comment line is processed in order; command can cancel each other (if you have both enable and disable commands)
   2. With semantic ordering: there is a prioritization between enabling and disabling (prioritization TBD) 
1. Favor only one Stryker comment
   1. The first one 
   3. The last one
   4. The most specific
   That being said, Stryker could log a warning regarding the discarded comments
1. Rejects all Stryker comments and log a warning 

Bearing in mind that there is valid use cases for having multiple Stryker comments (for a single Syntax Node), the logical conclusion is to maintain the current logic but with an added warning in the log file for information.

The single real world situation (I can think of) where more than one Stryker comment is present is as a consequence of a code merge.
 
Aggregation requires implementing the aggregation logic (sequentially processing the command will not provide the expected result in the general case).
Rejecting all Stryker comments looks a bit drastic.

If valid use cases for multiple comments are identified, I am pretty sure the appropriate logic will be identified as well.

Reply:
Understood, thank you for the clarification. I think the warnings you introduced add a lot of clarity to the current implementation.

Perhaps a more realistic scenario where multiple Stryker comments might be used would be if someone needs to use a combination of `once` and non-`once` comments, as they can't be merged together (at least as far as I'm aware). For example:
```csharp
int SomeMethod(bool flagA, bool flagB, int y, int z) {
  // Stryker disable Arithmetic
  // Stryker disable once all
  var x = (flagA && flagB) ? 10 : (y + z); // Should have all mutations disabled.
  return x + 5; // Should only have Arithmetic mutations disabled.
}
```

If that's a scenario we want to support, perhaps we should only allow at most a single non-`once` comment and a single `once` comment on any given node. It might also make sense to require the `once` comment to come *after* the non-`once` comment, both for clarity and to avoid scope-related issues.

Reply:
For the time being, let's keep it simple. There are several ways to achieve this granularity without using multiple Stryker comments (on the same construct).

Since you want to disable arithmetic for the whole method, why not place it at the method level
```csharp
// Stryker disable Arithmetic
int SomeMethod(bool flagA, bool flagB, int y, int z) {
  // Stryker disable once all
  var x = (flagA && flagB) ? 10 : (y + z); // Should have all mutations disabled.
  return x + 5; // Should only have Arithmetic mutations disabled.
}
```
You could also place it at the block level using multiline comments.
Alternatively, only the first line requires specific handling, so you can disable arithmetic afterward:
```csharp
int SomeMethod(bool flagA, bool flagB, int y, int z) {
  // Stryker disable once all
  var x = (flagA && flagB) ? 10 : (y + z); // Should have all mutations disabled.
 // Stryker disable Arithmetic
  return x + 5; // Should only have Arithmetic mutations disabled.
}
```

or even inject an empty statement
```csharp
int SomeMethod(bool flagA, bool flagB, int y, int z) {
  /* Stryker disable Arithmetic */;
  // Stryker disable once all
  var x = (flagA && flagB) ? 10 : (y + z); // Should have all mutations disabled.
  return x + 5; // Should only have Arithmetic mutations disabled.
}
```
which is ugly, but does the job if you cannot use an alternative approach.

Regarding command aggregation, note that people could argue for that interpretation
```csharp
  // Stryker disable all
  // Stryker disable Arithmetic all
  here, only arithmetic mutations are ignored, the others are handled normally
  ```
 
The intent of Stryker comments is to allow user to manually ignore mutations that are semantically equivalent to the original code, or can not be tested due to some project constraint.
This way, they can sill target a 100% kill rate. But the objective never was to support surgical precision, that would allow users to target the exact mutations they want to disable.
Coder should favor the `disable once all` in general most of the time, using specific mutator selection when they do need other mutations to be tested. 
And it should be avoided and used as rarely as possible. When a codebase has a lot of Stryker comments, it will be often easier to disable some mutations than it is to kill it and then the interest of mutation testing is lost.

A feature being implementable and consistent with the existing design is not sufficient to make it a good idea 😄 .

Of course, this is only my opinion, and this can be discussed. But, as a last thing, I would add we can wait a few months to see what are the real use cases for this, if any arises